core/ledger/kvledger/txmgmt/txmgr/lockbasedtxmgr/tx_simulator.go

```diff
@@ -16,25 +16,25 @@ import (
 	"github.com/pkg/errors"
 )
 
-// LockBasedTxSimulator is a transaction simulator used in `LockBasedTxMgr`
-type lockBasedTxSimulator struct {
-	lockBasedQueryExecutor
+// TxSimulator is a transaction simulator used in `LockBasedTxMgr`
+type TxSimulator struct {
```

Comment:
Unexported struct implementing an exported function looks weird. Some discussion on this happened in #1094 Hence, I exported the struct. 

Btw, the initial goal was to first merge #1094, second merge #1095, and finally make the following change in this PR as mentioned in the PR description. `NewTxSimulation()` and `NewQueryExecutor()` to return the struct instead of an interface as it is a widely suggested way when there is a single implementation for an interface. Now, the above-mentioned change will be done once the post-order tx execution is moved from validator to the `lockbasedtxmgr` and the `TxMgr` interface is removed.